internal/rpc/rpc.go

```diff
@@ -240,7 +240,9 @@ func RetryStreamingServerStream[Req, Resp any](
 				} else {
 					// Stream terminated; check if this was caused by an error
 					err = stream.Err()
-					logLevel = log.Warn
+					if !useDebugErrorLevel(err) {
```

Comment:
For readability, could we refactor this to `logLevel = logLevelForError(err)`? That way, you don't need to mentally parse the `!` and the variable scope/flow.